File: Exercism/csharp/perfect-numbers/PerfectNumbers.cs
using System;
using System.Linq;

public enum Classification
{
    Perfect = 0,
    Abundant = 1,
    Deficient = -1
}

public static class PerfectNumbers
{
    public static Classification Classify(int number) => number < 1
      ? throw new ArgumentOutOfRangeException()
      : (Classification)Enumerable.Range(1, number / 2).Where(i => number % i == 0).Sum().CompareTo(number);
}

File: Exercism/fsharp/perfect-numbers/PerfectNumbers.fs
module PerfectNumbers

type Classification =
    | Perfect
    | Abundant
    | Deficient

let classify =
    function
    | n when n < 1 -> None
    | n ->
        [ for i in 1 .. n / 2 do if n % i = 0 then i ]
        |> Seq.sum
        |> (function
        | s when s < n -> Deficient
        | s when s > n -> Abundant
        | _ -> Perfect)
        |> Some
